src/main/java/fi/aalto/cs/apluscourses/intellij/actions/TutorialAction.java

```diff
@@ -90,6 +96,21 @@ private void doTutorial(@NotNull Project project) {
     }
     TutorialExercise tutorialExercise = (TutorialExercise) selectedExercise.getModel();
 
+    Set<String> dependencies = tutorialExercise.getTutorial().getDependencies();
+    Set<String> missingDependencies = new java.util.HashSet<>();
```

Comment:
It feels weird to see a fully qualified default namespace `java.util` :p Perhaps add it to imports?

Reply:
I stand up for the "alternative" way Taikelenn described. Loops and mutable collections are something to be avoided whenever possible.